### ISSUE TITLE
It's not obvious you need to hit enter to save a dynamic filter rule

### ISSUE THREAD

**Issue by unixandria-xda** [uBlockOrigin/uBlock-issues#1245]
### Prerequisites

- [x] I verified that this is not a filter issue
    - Filter issues MUST be reported at [filter issue tracker](https://github.com/uBlockOrigin/uAssets/issues)
    
- [x] This is not a support issue or a question
    - Support issues and questions are handled at [/r/uBlockOrigin](https://old.reddit.com/r/uBlockOrigin/)
    
- [x] I performed a cursory search of the issue tracker to avoid opening a duplicate issue
    - Your issue may already be reported.
- I tried to reproduce the issue when...
    - [x] uBlock Origin is the only extension
    - [x] uBlock Origin with default lists/settings
    - [x] using a new, unmodified browser profile
- [x] I am running the latest version of uBlock Origin
- [x] I checked the [documentation](https://github.com/gorhill/uBlock/wiki) to understand that the issue I report is not a normal behavior

### Description

When creating a dynamic filtering rule such as `* * 3p-frame block` if you do not hit enter after inputting that it is silently discarded, with no warning to the user.

### A specific URL where the issue occurs

n/a

### Steps to Reproduce

1. Go to "my rules" pane
2. Input `* * 3p-frame block`
3. HIt save, and see nothing is saved.

### Expected behavior:

Either a text on the page saying you must hit enter after every rule, including the last one, or waiving the requirement to hit enter on the last line

### Actual behavior:

If you don't hit enter the last rule is discarded silently.

### Your environment

* uBlock Origin version: 1.29.2
* Browser Name and version: Firefox 82.0a1
* Operating System and version: Windows 10, 2004

**Comment by uBlock-user** [comment]
>  see nothing is saved.

<kbd>Commit</kbd> button gets active and now you need to click <kbd>Commit</kbd>, to get it saved to permanent rules.

TLDR: Hit <kbd>Save</kbd>, then click <kbd>Commit</kbd>.

**Comment by gorhill** [comment]
> HIt save, and see nothing is saved

I hit _Save_ and the rule was saved. Unable to reproduce.

**Comment by unixandria-xda** [comment]
Save for me _always_ discards any edits to the last line. Like it doesnt even show up under temp rules

**Comment by gorhill** [comment]
Please make screenshots for:
- Before entering edit mode
- After pasting the rule, before clicking Save
- After exiting edit mode once Save is clicked

**Comment by unixandria-xda** [comment]
This issue is only reproducible on mobile, firefox nightly

![Screenshot_20200915-141523814](https://user-images.githubusercontent.com/45006100/93248533-f5700580-f75d-11ea-9afa-148056449a57.jpg)
![Screenshot_20200915-141516685](https://user-images.githubusercontent.com/45006100/93248535-f6a13280-f75d-11ea-8035-795c6e168d0f.jpg)
![Screenshot_20200915-141456211](https://user-images.githubusercontent.com/45006100/93248537-f6a13280-f75d-11ea-9b22-4a5d2dd09666.jpg)

**Comment by unixandria-xda** [comment]
Goes from before -> editing -> after save, reverse order bottom to top

**Comment by gorhill** [comment]
So you were using Firefox for Android, while you reported the issue was with Firefox Desktop.

There are known issues with various CodeMirror + some Android keyboards.

It works fine on my side using [AnySoftKeyboard](https://github.com/AnySoftKeyboard/AnySoftKeyboard) with composition disabled -- the issue is probably that your keyboard is in composition mode as seen in your screenshot, so this means the editor did not yet receive the text change. Just force exit composition by selecting the `block` suggestion, or use another keyboard software.

**Comment by unixandria-xda** [comment]
I mean google keyboard is the default, so having behavior that conflicts with default software is rather undesirable.

**Comment by gorhill** [comment]
Workaround, hit <enter>.

**Comment by unixandria-xda** [comment]
I was originally suggesting to put a tooltip that you do have to do that

Eg
"On Android some keyboards don't send text input correctly, so you'll need to hit enter after the last rule to get it to save"

Also maybe don't silently discard invalid input, lol

**Comment by gorhill** [comment]
> On Android some keyboards don't send text input correctly, so you'll need to hit enter after the last rule to get it to save

You are the only one who complained about this so far, so I am not going to add a highly annoying and long warning on a space limited display. You even found the workaround on your own.

See: https://github.com/codemirror/CodeMirror/issues/5244
